rancher/webhook
# Add Validation on NodeDriver update/delete to prevent machine deletion

## Issue: https://github.com/rancher/rancher/issues/41128 

 
## Problem


Currently there is no validation on the `NodeDriver` object and when a node driver is disabled/deleted it currently removes the related CRDs which can result in machines being deleted and data loss.
 
## Solution

By adding a webhook for the `NodeDriver` resource we can validate that there are 0 rke1/rke2 machines using that driver before allowing the request to go through.
 
## Testing

1. Create a cluster with a node driver (digitalocean is what i used)
2. Go to the drivers page and try and disable the driver -> it will fail after this patch. 
3. Remove the cluster
4. Disabling the driver will go through as expected.
